src/components/pages/add-revision.tsx

```diff
@@ -95,6 +97,41 @@ export function AddRevision({
       </StaticInfoPanel>
     )
 
+  // types needs refactoring here. splitting controls and data would probably make sense
+
+  const onSave = async (
+    data:
+      | SetEntityMutationData
+      | PageSerializedState
+      | TaxonomyCreateOrUpdateMutationData
+  ) => {
+    const willNeedReview = hasOwnPropertyTs(data, 'controls')
+      ? !(data as OnSaveData).controls.noReview
+      : entityNeedsReview
+
+    const success =
+      type === UuidType.Page
+        ? await addPageRevision(data as PageSerializedState)
+        : type === UuidType.TaxonomyTerm
+        ? await taxonomyCreateOrUpdateMutation(
+            data as TaxonomyCreateOrUpdateMutationData
+          )
+        : await setEntityMutation(
+            {
+              ...data,
+              __typename: type,
+            } as SetEntityMutationData,
+            willNeedReview,
+            initialState,
+            taxonomyParentId
+          )
```

Comment:
just a note: here I see the potential to combine all endpoints into one... would be a bigger refactoring step though...